File: server/src/main/java/br/com/celk/service/UnidadeFederativaService.java
package br.com.celk.service;

import br.com.celk.domain.UnidadeFederativa;
import br.com.celk.repository.UnidadeFederativaRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Optional;

@Service
public class UnidadeFederativaService implements Serializable {
    private final UnidadeFederativaRepository repository;

    public UnidadeFederativaService(UnidadeFederativaRepository repository) {
        this.repository = repository;
    }

    public Page<UnidadeFederativa> listAll(Pageable pageable) {
        return repository.findAll(pageable);
    }

    public UnidadeFederativa findById(Long id) {
        return repository.findById(id).orElse(new UnidadeFederativa());
    }

    public UnidadeFederativa save(UnidadeFederativa entity) {
        entity.setDataAtualizacao(LocalDateTime.now());
        return repository.save(entity);
    }

    public void delete(Long id) {
        Optional<UnidadeFederativa> entity = repository.findById(id);
        entity.ifPresent(repository::delete);
    }

    public Page<UnidadeFederativa> listAllByFiltro(String search, Pageable pageable) {
        return repository.findAllByNomeContainsIgnoreCaseOrSiglaContainsIgnoreCase(search, search, pageable);
    }
}

File: web-app/src/services/UnidadeFederativaService.ts
import { AxiosService } from './AxiosService';

const URL = '/uf';

const listAll = async (path: string, search: string, page?: string) => {
    return AxiosService.get(`${URL}${path}?search=${search ? search : ''}&${page}`);
};

const save = async (data: any) => {
    return AxiosService.post(`${URL}`, data);
};

const deletar = async (id: number) => {
    return AxiosService.delete(`${URL}/${id}`);
};

export const UnidadeFederativaService = {
    listAll,
    save,
    deletar,
};
